modules/plugin-search.php

```diff
@@ -0,0 +1,226 @@
+					'version' => '',
+					'author' => '',
+					'author_profile' => '',
+					'requires' => '',
+					'tested' => '',
+					'requires_php' => '',
+					'rating' => 100,
+					'ratings' => array('1'=>1,'2'=>2,'3'=>3,'4'=>4,'5'=>5),
+					'num_ratings' => 100,
+					'support_threads' => 100,
+					'support_threads_resolved' => 100,
+					'active_installs' => 3000000,
+					'downloaded' => 10000000,
+					'last_updated' => '',
+					'added' => '',
+					'homepage' => '',
+					'download_link' => '',
+					'tags' => array(),
+					'donate_link' => '',
+					'short_description' => '',
+					'description' => '',
+					'icons' => array(
+						'1x'  => 'https://ps.w.org/jetpack/assets/icon.svg?rev=1791404',
+						'2x'  => 'https://ps.w.org/jetpack/assets/icon-256x256.png?rev=1791404',
+						'svg' => 'https://ps.w.org/jetpack/assets/icon.svg?rev=1791404',
+					),
+				);
+
+				// Splice in the base module data
+				$inject = array_merge( $inject, $jetpack_modules_list[ $matching_module ] );
+
+				// Helps to determine if that an injected card.
+				$inject['plugin-search'] = true;
+				// Supplement name/description so that they clearly indicate this was added.
+				$inject['name'] = sprintf(
+					__( 'Jetpack: %s', 'Jetpack: Module Name' ),
+					$jetpack_modules_list[ $matching_module ]['name']
+				);
+				$inject['short_description'] = sprintf(
+					__( 'You already have Jetpack installed, and it provides this functionality. %s', 'You already have Jetpack installed... Module description.' ),
+					$jetpack_modules_list[ $matching_module ]['short_description']
+				);
+
+				// Add it to the top of the list
+				array_unshift( $result->plugins, $inject );
+			}
+		}
+		return $result;
+	}
+
+	/**
+	 * Take a raw search query and return something a bit more standardized and
+	 * easy to work with.
+	 *
+	 * @param  String $term The raw search term
+	 * @return String A simplified/sanitized version.
+	 */
+	private function sanitize_search_term( $term ) {
+		$term = strtolower( urldecode( $term ) );
+
+		// remove non-alpha/space chars.
+		$term = preg_replace( '/[^a-z ]/', '', $term );
+
+		// remove strings that don't help matches.
+		$term = trim( str_replace( array( 'jetpack', 'jp', 'free', 'wordpress' ), '', $term ) );
+
+		return $term;
+	}
+
+	/**
+	 * Tracks every search term used in plugins search as 'jetpack_wpa_plugin_search_term'
+	 *
+	 * @param String $term The raw search term.
+	 * @return true|WP_Error true for success, WP_Error if error occurred.
+	 */
+	private function track_search_term( $term ) {
+		return JetpackTracking::record_user_event( 'wpa_plugin_search_term', array( 'search_term' => $term ) );
+	}
+
+	/**
+	 * Callback function to sort the array of modules by the sort option.
+	 */
+	private function by_sorting_option( $m1, $m2 ) {
+		return $m1['sort'] - $m2['sort'];
+	}
+
+	/**
+	 * Put some more appropriate links on our custom result cards.
+	 */
+	public function insert_module_related_links( $links, $plugin ) {
```

Comment:
We should make sure to add Tracks to these links, so that we understand how people are using it/which links are most useful, and probably also something that indicates whenever one of these cards/results are shown.